src/operation/watch/test.rs

```diff
@@ -0,0 +1,221 @@
+use std::time::Duration;
+
+use crate::{
+    bson::{doc, Document},
+    bson_util,
+    change_stream::ChangeStreamTarget,
+    cmap::{CommandResponse, StreamDescription},
+    operation::{Operation, Watch},
+    options::{
+        ChangeStreamOptions,
+        FullDocumentType,
+        ReadPreference,
+        SelectionCriteria,
+        StreamAddress,
+    },
+    Namespace,
+};
+
+fn build_test(
+    target: impl Into<ChangeStreamTarget>,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    mut expected_body: Document,
+) {
+    let watch = Watch::new(target.into(), pipeline, options).unwrap();
+
+    let mut cmd = watch.build(&StreamDescription::new_testing()).unwrap();
+
+    assert_eq!(cmd.name.as_str(), "aggregate");
+
+    bson_util::sort_document(&mut expected_body);
+    bson_util::sort_document(&mut cmd.body);
+
+    assert_eq!(cmd.body, expected_body);
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build() {
```

Comment:
can we update the spacing in all these `doc!` invocations to be consistent? (i.e. spaces after every `[` or `{` and before every `]` or `}`). You don't need them after at the beginning and end of `vec!` invocations though, and rustfmt will enforce that I think.

Reply:
fixed

Reply:
There are still many in this file that need proper spacing. Also it's not necessary to invoke `doc!` from within itself.
e.g. on line 51:
```rust
"pipeline": [doc! {"$changeStream": {"fullDocument": "updateLookup"}}, doc! { "$match": { "x": 3 }}]
// should be
"pipeline": [{ "$changeStream": { "fullDocument": "UpdateLookup" } }, { "$match": { "x": 3 } }]
```

These often will look a lot better if you add line breaks.
```rust
"pipeline": [
    { "$changeStream": { "fullDocument": "UpdateLookup" } },
    { "$match": { "x": 3 } }
]
```
Check out [`src/operation/find/test.rs`](https://github.com/mongodb/mongo-rust-driver/blob/master/src/operation/find/test.rs) for examples of some well-formatted `doc!` invocations.

Reply:
The example I referred to still does not appear to be completely fixed, in addition to several other instances in this file. Refer to the examples I proved above and Isabel's comment for more information on what we're looking for.